1/contrib/s2i/run

```diff
@@ -9,11 +9,36 @@
 source /usr/local/bin/jenkins-common.sh
 source /usr/local/bin/kube-slave-common.sh
 
+#get the fully qualified paths to both 32 and 64 bit java
+JVMPath32bit=`alternatives --display java | grep family | grep i386 | awk '{print $1}'`
+JVMPath64bit=`alternatives --display java | grep family | grep x86_64 | awk '{print $1}'`
+
+#if machine has 4GB or less, going to force use of 32bit to save space unless user
+#specifically want to force 64bit
+CONTAINER_MEMORY_IN_BYTES=`cat /sys/fs/cgroup/memory/memory.limit_in_bytes`
+MEMORY_LIMIT_FOR_32BIT=$((2**32-1))
+
+# set the java version used based on OPENSHIFT_JENKINS_JVM_ARCH
+if [ -z $OPENSHIFT_JENKINS_JVM_ARCH  ]; then
+    if [ "${CONTAINER_MEMORY_IN_BYTES}" -gt "${MEMORY_LIMIT_FOR_32BIT}" ]; then
```

Comment:
unfortunately it's more complicated than this.

What the check really needs to be based on is whether the computed heap size is greater than what a 32 bit jvm allows.  Depending on CONTAINER_HEAP_PERCENT, the computed heap could be significantly less than the container memory.

also what is the actual max value a 32 bit jvm on linux allows the heap to be set to?  is it 2**32-1?  i would think it might be a bit lower than that since some process memory needs to be reserved for native usage.